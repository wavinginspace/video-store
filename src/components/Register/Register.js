import React, { Component } from 'react';
import './Register.scss';

export class Register extends Component {
  render() {
    return (
      <>
        <h2>Register</h2>

        <div className="Register box">
          <form action="" name="register-form" className="register-form">
            <label htmlFor="email">Email:</label>
            <input type="email" name="email" className="register-input" />

            <label htmlFor="email">Password:</label>
            <input type="password" name="password" className="register-input" />
            <label htmlFor="email">Confirm Password:</label>
            <input
              type="password"
              name="passwordconfirm"
              className="register-input"
            />
            <button type="submit">Submit</button>
          </form>
        </div>
      </>
    );
  }
}

export default Register;
