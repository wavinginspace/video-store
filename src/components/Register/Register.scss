.Register {
  margin: 0 auto;
}

.register-form {
  display: flex;
  justify-content: center;
  flex-direction: column;
}

h2 {
  text-align: center;
}

