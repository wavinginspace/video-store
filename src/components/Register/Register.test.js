import React from 'react';
import ReactDOM from 'react-dom';
import Register from './Register.test';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<Register />, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer.create(<Register />).toJSON();
  expect(tree).toMatchSnapshot();
});
