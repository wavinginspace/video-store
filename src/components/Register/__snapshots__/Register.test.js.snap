// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <h2>
    Register
  </h2>,
  <div
    className="Register box"
  >
    <form
      action=""
      className="register-form"
      name="register-form"
    >
      <label
        htmlFor="email"
      >
        Email:
      </label>
      <input
        className="register-input"
        name="email"
        type="email"
      />
      <label
        htmlFor="email"
      >
        Password:
      </label>
      <input
        className="register-input"
        name="password"
        type="password"
      />
      <label
        htmlFor="email"
      >
        Confirm Password:
      </label>
      <input
        className="register-input"
        name="passwordconfirm"
        type="password"
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>,
]
`;
